python/pyspark/rdd.pyi

```diff
@@ -445,11 +445,18 @@ class RDD(Generic[T]):
     @overload
     def toDF(
         self: RDD[RowLike],
-        schema: Optional[List[str]] = ...,
+        schema: Optional[Union[List[str], Tuple[str, ...]]] = ...,
         sampleRatio: Optional[float] = ...,
     ) -> DataFrame: ...
     @overload
-    def toDF(self: RDD[RowLike], schema: Optional[StructType] = ...) -> DataFrame: ...
+    def toDF(
+        self: RDD[RowLike], schema: Optional[Union[StructType, str]] = ...
+    ) -> DataFrame: ...
+    @overload
+    def toDF(
+        self: RDD[Union[DateTimeLiteral, LiteralType, DecimalLiteral]],
```

Comment:
It seems like this part will requires some changes (switch to `TypeVar`). I'll investigate this and provide an update.